### PR TITLE
fix invalid contact

### PR DESCRIPTION
**Pull request by lazedo**

### CODE REVIEW

File: applications/ecallmgr/src/ecallmgr_fs_notify.erl [hunk 1]
```diff
@@ -256,7 +256,15 @@ register_overwrite(JObj, Props) ->
                  ]).
 
 -spec ensure_contact_user(ne_binary(), ne_binary(), ne_binary()) -> api_binary().
-ensure_contact_user(Contact, Username, Realm) ->
+ensure_contact_user(OriginalContact, Username, Realm) ->
+    Contact = case binary:split(OriginalContact, <<";">>, ['global']) of
+                  [<<>> | L] ->
+                      lists:foldl(fun(A,B) ->
+                                          <<B/binary, ";", A/binary>>
+                                  end, <<"sip:", Username/binary, "@", Realm/binary>>, L);
```
Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_fs_notify.erl]: If you're trying to just remove the `$;` prefix from the binary, why not just pattern-matching it?
You're deconstructing `OriginalContact` based on `;` then re-assembling it again, right?

File: applications/ecallmgr/src/ecallmgr_registrar.erl [hunk 2]
```diff
@@ -888,14 +888,33 @@ augment_registration(Reg, JObj) ->
...
+                                 ,['global']
+                                ),
+    Contact = case binary:split(TempContact, <<";">>, ['global']) of
+                  [<<>> | L] ->
+                      lists:foldl(fun(A,B) ->
+                                          <<B/binary, ";", A/binary>>
+                                  end, <<"sip:", Username/binary, "@", Realm/binary>>, L);
+                  _L ->
+                      OriginalContact
+              end,
```
Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_registrar.erl]: Same here

File: applications/ecallmgr/src/ecallmgr_fs_notify.erl (hunk 1 above)
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_notify.erl]: @fenollp show me. try these "sip:sss;fs_path=fff", "@123.4.5.6;fs_path=..", ";fs_path=..."

Comment (jamesaimonetti) [inline applications/ecallmgr/src/ecallmgr_fs_notify.erl]: Sounds like an opportunity for some unit tests!

Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_fs_notify.erl]:
```
case OriginalContact of
    <<$;, _/binary>> -> <<"sip:", Username/binary, "@", Realm/binary, OriginalContact/binary>>;
    _ -> OriginalContact
```

File: applications/ecallmgr/src/ecallmgr_util.erl [hunk 3]
```diff
@@ -1146,3 +1148,25 @@ get_dial_separator(JObj, Endpoints) ->
...
                        ,Endpoints
                       ).
+
+-spec fix_contact(api_binary() | list(), ne_binary(), ne_binary()) -> api_binary().
+fix_contact('undefined', _, _) -> 'undefined';
+fix_contact(<<";", _/binary>> = OriginalContact, Username, Realm) ->
+    fix_contact(<<"sip:", Username/binary, "@", Realm/binary, OriginalContact/binary>>, Username, Realm);
+fix_contact(OriginalContact, Username, Realm)
+  when is_binary(OriginalContact) ->
+    fix_contact(binary:split(wh_util:strip_binary(OriginalContact), <<";">>, ['global']), Username, Realm);
```
Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_util.erl]: Is it possible that `OriginalContact` is prefixed by more than one `;`?

File: applications/ecallmgr/src/ecallmgr_util.erl [hunk 4]
```diff
@@ -1146,3 +1148,25 @@ get_dial_separator(JObj, Endpoints) ->
     get_dial_separator(wh_json:get_value(<<"Dial-Endpoint-Method">>, JObj, ?DIAL_METHOD_SINGLE)
                        ,Endpoints
                       ).
+
+-spec fix_contact(api_binary() | list(), ne_binary(), ne_binary()) -> api_binary().
+fix_contact('undefined', _, _) -> 'undefined';
+fix_contact(<<";", _/binary>> = OriginalContact, Username, Realm) ->
+    fix_contact(<<"sip:", Username/binary, "@", Realm/binary, OriginalContact/binary>>, Username, Realm);
```
Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_util.erl]: Didn't you forget a `;` between `Realm` and `OriginalContact`?

Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_util.erl]: no

File: applications/ecallmgr/src/ecallmgr_util.erl (hunk 3 above)
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_util.erl]: yes

**Pull request merged**